src/pkg/transform/image.go

```diff
@@ -63,39 +63,53 @@ func ImageTransformHostWithoutChecksum(targetHost, srcReference string) (string,
 
 // ParseImageRef parses a source reference into an Image struct
 func ParseImageRef(srcReference string) (out Image, err error) {
-	ref, err := reference.ParseAnyReference(srcReference)
-	if err != nil {
-		return out, err
-	}
 
-	// Parse the reference into its components
-	if named, ok := ref.(reference.Named); ok {
-		out.Name = named.Name()
-		out.Path = reference.Path(named)
-		out.Host = reference.Domain(named)
-		out.Reference = ref.String()
+	if IsTarball(srcReference) {
+		out.Host = "docker.io"
+		out.Name = srcReference
+		out.Path = fmt.Sprintf("%s/%s/%s", out.Host, "library", strings.TrimSuffix(srcReference, ".tar"))
+		out.Reference = srcReference
```

Comment:
Editing the reference like this won't result in a proper image reference in the end - while the package will create,  the `org.opencontainers.image.base.name` remains `"zarf-game.tar"` inside `images/index.json` within the zarf package.
![image](https://github.com/defenseunicorns/zarf/assets/3977569/6855b309-d5ab-4eb2-a081-a65062e0a767)

This is loaded on deploy successfully but then when pushed converts into `docker.io/library/zarf-game` which does not match the `defenseunicorns/zarf-game:multi-tile-dark` the tarball was made from.  This means that the Deployment will never pull the image and will be in an image pull backoff state:
![image](https://github.com/defenseunicorns/zarf/assets/3977569/90ea208d-c642-4bef-96fe-297d19428e1d)

Reply:
What needs to happen is that the image reference in the tarball needs to be parsed from it's manifest.json:
![image](https://github.com/defenseunicorns/zarf/assets/3977569/c00d9320-7bf1-4bcd-b8c8-26392c3dfdff)
![image](https://github.com/defenseunicorns/zarf/assets/3977569/625816b2-4190-4fd8-b24a-6daab820a3fd)
(you can see this if you vim into it, and it is accessible when the tarball is loaded with crane)

Reply:
For this, the `zarf.yaml` image reference may need to be modified to turn `zarf-game.tar` into `defenseunicorns/zarf-game:multi-tile-dark` after pulling it